examples/lighting-app/silabs/src/ZclCallbacks.cpp

```diff
@@ -60,8 +62,66 @@ void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &
     {
         ChipLogProgress(Zcl, "Color Control attribute ID: " ChipLogFormatMEI " Type: %u Value: %u, length %u",
                         ChipLogValueMEI(attributeId), type, *value, size);
+// WIP Apply attribute change to Light
+#if (defined(SL_MATTER_RGB_LED_ENABLED) && SL_MATTER_RGB_LED_ENABLED == 1)
+        EndpointId endpoint = attributePath.mEndpointId;
+        ColorData_t colorData;
+        /* XY color space */
+        if (attributeId == ColorControl::Attributes::CurrentX::Id || attributeId == ColorControl::Attributes::CurrentY::Id)
```

Comment:
I don't think this is a good approach.
Rather than receiving 1 attribute change update, then manually reading the counterpart. LightManager should probably keep a ram copy of the last update like we do for OnOff.

Same thing for CurrentHue and CurrentSaturation.
essentially this becomes.

1. Receive MatterPostAttributeChangeCallback
2. Update the LightManager ram copy
3. Post-action to update the physical RGBLED

Reply:
Updated code accordingly to your comment.